src/common/secret.c

```diff
@@ -107,7 +121,7 @@ int get_secret_option(const char *secret, const char *key_name,
   if (secret) {
     ret = set_kernel_secret(secret, key_name);
     if (ret < 0) {
-      if (ret == -ENODEV || ret == -ENOSYS) {
+      if (ret == -ENODEV || ret == -ENOSYS || ret == -EINVAL) {
```

Comment:
@tchaikov 
I'll return ENOSYS.
And one of the reasons to implement this, is that will allow parts of the FreeBSD variants of mount.ceph and krbd to share more of the common code.
Next to the fact that it now will allow to make less exemptions in the Cmake files. Building a library that includes secret.c.o (for what ever reason) will no longer fail now.